test/test_model.py

```diff
@@ -1105,7 +1111,13 @@ def assertBackendCalls(  # noqa: N802
         assert expected == harness._get_backend_calls(reset=reset)
 
     def test_run_error(self, fake_script: FakeScript):
-        model = ops.Model(ops.CharmMeta(), _ModelBackend('myapp/0'))
+        model = ops.Model(
+            ops.CharmMeta(),
+            _ModelBackend(
+                juju_context=ops.main._JujuContext.from_environ(dict(os.environ)),
+                unit_name='myapp/0',
+            ),
+        )
```

Comment:
This feels like a code smell.

If the test expects specific values in the environment, let's pass these values here, like `_ModelBackend(juju_context=_JujuContext("foo", "bar", other="baz"))`.

If the test doesn't expect specific values, it's better to supply a blank context.

If the test depends parent process environment, that's probably a bug.

Reply:
I agree; this is solved by the previous comment from Ben, though. We are making `juju_context` optional with a default value coming from `os.environ`.